keycache/keycache.go

```
@@ -94,6 +95,24 @@ func NewCache() Cache {
 	return Cache{make(map[DelegateIndex]ActiveUser)}
 }
 
+// NewFrom takes the output of GetSummary and returns a new keycache.
+func NewFrom(summary map[string]ActiveUser) *Cache {
+	cache := &Cache{make(map[DelegateIndex]ActiveUser)}
+	for di, user := range summary {
+		diSplit := strings.SplitN(di, "-", 2)
+		index := DelegateIndex{
+			Name: diSplit[0],
+		}
+
+		if len(diSplit) == 2 {
+			index.Slot = diSplit[1]
+		}
```

Are there other cases here that should return an error?